File: src/lib/tauri/serverProfiles.ts
import { invoke } from "@tauri-apps/api/core";
import type {
  SaveServerProfileInput,
  ServerProfileSummary
} from "../../types/serverProfiles";

export async function listServerProfiles(): Promise<ServerProfileSummary[]> {
  return invoke<ServerProfileSummary[]>("list_server_profiles");
}

export async function saveServerProfile(
  input: SaveServerProfileInput
): Promise<ServerProfileSummary> {
  return invoke<ServerProfileSummary>("save_server_profile", { input });
}

File: src-tauri/src/commands/server_profiles.rs
use crate::{app_state::AppState, modules::models::{SaveServerProfileInput, ServerProfileSummary}};
use tauri::State;

#[tauri::command]
pub fn save_server_profile(
    input: SaveServerProfileInput,
    state: State<'_, AppState>,
) -> Result<ServerProfileSummary, String> {
    state
        .database
        .save_server_profile(input)
        .map_err(|error| error.to_string())
}

#[tauri::command]
pub fn list_server_profiles(
    state: State<'_, AppState>,
) -> Result<Vec<ServerProfileSummary>, String> {
    state
        .database
        .list_server_profiles()
        .map_err(|error| error.to_string())
}
